sql/plan/truncate.go

```diff
@@ -154,6 +154,12 @@ func (p *Truncate) WithChildren(children ...sql.Node) (sql.Node, error) {
 	return &nt, nil
 }
 
+// CheckPrivileges implements the interface sql.Node.
+func (p *Truncate) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	return opChecker.UserHasPrivileges(ctx,
+		sql.NewPrivilegedOperation(p.db, getTableName(p.Child), "", sql.PrivilegeType_Drop))
```

Comment:
This isn't just DELETE?

Reply:
Seems like it should be, but perhaps it's closer to a `DROP TABLE` internally so they rely on `DROP` instead of `DELETE`.